File: Anul II/Medii de proiectare si programare/Laborator/lab02/C#/Ubb.BikeContest.Model/src/RaceEntry.cs
namespace Ubb.BikeContest.Model;

public class RaceEntry : Identifiable<long>
{
    private readonly Participant _participant;
    private readonly Race _race;

    public RaceEntry(long id, Participant participant, Race race)
    {
        Id = id;
        _participant = participant;
        _race = race;
    }

    public Participant Participant => _participant;

    public Race Race => _race;
}

File: Anul II/Medii de proiectare si programare/Laborator/lab11/java/model/src/main/java/com/ubb/model/RaceEntry.java
package com.ubb.model;

import java.util.Objects;

public class RaceEntry extends Identifiable<Long> {
    private final Participant participant;
    private final Race race;

    {
        setId(0L);
    }

    public RaceEntry(Participant participant, Race race) {
        this.participant = participant;
        this.race = race;
    }

    public Participant getParticipant() {
        return participant;
    }

    public Race getRace() {
        return race;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        RaceEntry raceEntry = (RaceEntry) o;
        return getParticipant().equals(raceEntry.getParticipant()) && getRace().equals(raceEntry.getRace());
    }

    @Override
    public int hashCode() {
        return Objects.hash(getParticipant(), getRace());
    }

    @Override
    public String toString() {
        return participant.getFirstName() + " " + participant.getLastName() + " registered for '"
                + race.getName() + " " + race.getEngineCapacity() + "cc'";
    }
}
